### PR TITLE
add exception for junit-pioneer in version rejection of snapshots

### PR DESCRIPTION
**Pull request by Siedlerchr**
I now added an exception for junit-pioneer  version checker. All other snapshots are rejected.

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

**Comment by Siedlerchr** [comment]
@LinusDietz with exception I mean that junit-pioneer does not get rejected, because it's only available as snapshot. Same applies to javax:inject

if you run ./gradlew dU it works correclty

File: build.gradle [hunk 1]
```diff
@@ -188,14 +188,19 @@ dependencyUpdates.revision = 'integration'
...
                 selection.reject("Ignore SNAPSHOT releases")
             }
         }
         rules.withModule("org.controlsfx:controlsfx") { ComponentSelection selection ->
             if (selection.candidate.version ==~ /9.*/) { // Reject version 9 or higher
                  selection.reject("Cannot be updated to 9.*.* until Jabref works with Java 9")
             }
+        }
+		rules.withModule("com.github.tomtung:latex2unicode_2.12") { ComponentSelection selection ->
+            if (selection.candidate.version ==~ /0.2.2/) { // Reject version higher than 2.0.2
```
Comment (LinusDietz) [inline build.gradle]: `selection.candidate.version == 0.2.4` would be correct. (you have to put in the version to be rejected, not the current.

File: build.gradle [hunk 2]
```diff
@@ -188,14 +188,19 @@ dependencyUpdates.revision = 'integration'
...
+                 selection.reject("Cannot be updated to 0.2.4 until JabRef is prepared for it")
```
Comment (LinusDietz) [inline build.gradle]: 0.2.4 -> 2.2.4

File: build.gradle [hunk 3]
```diff
@@ -188,14 +188,19 @@ dependencyUpdates.revision = 'integration'
 dependencyUpdates.resolutionStrategy = {
     componentSelection { rules ->
         rules.all { ComponentSelection selection ->
-            if (selection.candidate.version ==~ /[0-9].*SNAPSHOT/) {
+            if ( selection.candidate.module != "junit-pioneer" && selection.candidate.module!="javax.inject"  && selection.candidate.version ==~ /[0-9].*SNAPSHOT/ ) {
```
Comment (LinusDietz) [inline build.gradle]: javax.inject -> javax.inject:javax.inject ?

Comment (Siedlerchr) [reply build.gradle]: I initially had the same idea but my tests with junit pioneer showed that it didn't work. I then added a print statement and saw that the module is just the real module name
```
 componentSelection { rules ->
        rules.all { ComponentSelection selection ->
print selection.candidate.module 
```

**Comment by Siedlerchr** [comment]
As this works as expected I merge this now,

**Pull request merged**